packages/util/src/types.ts

```diff
@@ -1,4 +1,4 @@
 /**
  * Represents a type that may or may not be a promise
  */
-export type Awaitable<T> = PromiseLike<T> | T;
+export type Awaitable<Result> = PromiseLike<Result> | Result;
```

Comment:
Wouldn't `Value` be more appropiate? As a `Promise` wraps a `Value` inside a result.